### PR TITLE
feat(streamlit): add live experiment workspace and results

### PR DESCRIPTION
**Pull request by liq22**
## Summary

Completes the optional PHM-Vibench Streamlit experience as a four-step workspace:

1. select a validated registry template;
2. edit catalog-approved fields or portable YAML;
3. validate and launch the public `main.py --config` contract;
4. inspect live status, logs, metrics, images, and artifacts.

## User-facing improvements

- polished responsive hero and guided four-step flow;
- Quick Start defaults to the repository-shipped CPU smoke template;
- Advanced mode preserves typed YAML and raw override precedence;
- Run / Cancel / Restart from an immutable run snapshot;
- independent two-second log/result refresh through `st.fragment`;
- Overview / Metrics / Artifacts / Logs result tabs;
- recent-run selector and downloadable YAML, manifest, logs, and small artifacts;
- historical `streamlit run streamlit_app.py` command routes to the maintained workspace.

## Architecture and compatibility

- does not modify `main.py`, pipelines, factories, or the five-block schema;
- never imports or calls a Pipeline function directly;
- launches argv lists with `shell=False`;
- applies machine-local configuration exactly once;
- includes the machine-local configuration fingerprint in the validated run signature;
- isolates process lifecycle in `run_service.py`;
- isolates bounded artifact parsing in `result_service.py`;
- isolates visual and live-run components in `ui_theme.py` / `ui_runtime.py`;
- preserves registry and field-catalog extension points;
- intentionally omits non-portable pause/resume and multi-user scheduling.

## Cross-platform process policy

- POSIX: new session, process-group SIGTERM, then SIGKILL after grace period;
- Windows: new process group, CTRL_BREAK/terminate, then kill;
- detached POSIX processes block a second managed run and are not killed by PID alone;
- durable run manifests are written atomically under `outputs/streamlit/<run_id>/`.

## Result safety

- scans only the run directory and configured output root;
- rejects filesystem/repository/home roots and repository parents;
- skips symbolic links;
- bounds depth, entries, files, metric bytes, and metric rows;
- malformed or missing optional metrics become warnings rather than UI failures.

## Validation performed

```bash
python -m py_compile apps/streamlit/*.py
PYTHONPATH=. python -m pytest -q \
  test/test_streamlit_run_service.py \
  test/test_streamlit_result_service.py \
  test/test_streamlit_ui_imports.py
```

Result for the new process/result/UI slice: **18 passed**.

The configuration/preflight slice merged in PR #44 had **28 focused tests passing** in the same isolated implementation workspace.

The following repository-wide gates remain listed for environments with a complete checkout and core training dependencies:

```bash
python -m scripts.validate_configs
python -m scripts.validate_docs
python -m pytest test/
python main.py --config configs/demo/00_smoke/dummy_dg.yaml --override trainer.num_epochs=1
```

## Reviewer notes

Detailed accepted findings are recorded in `apps/streamlit/REVIEW.md`. The final review also added optional-Streamlit import coverage and invalidates preflight when `configs/local/local.yaml` changes.

### CODE REVIEW

**Review by liq22 — COMMENTED** [review]
Final compatibility/extensibility review completed.

Checked:
- public CLI boundary and absence of direct Pipeline imports;
- portable YAML/local-config/override precedence;
- validation-signature invalidation when machine-local config changes;
- argv-only subprocess execution and cross-platform process-group cancellation;
- atomic run manifests, restart provenance, detached-process handling;
- bounded artifact scanning, symlink rejection, malformed metric tolerance;
- optional Streamlit import wiring and legacy launcher compatibility;
- declarative field/registry extension points and separation of UI, process, and result layers.

Review-driven fixes included the local-config fingerprint and `test_streamlit_ui_imports.py`. No blocking review threads remain. The connector reports no CI status contexts for this head; repository-wide core/training gates remain explicitly documented in the PR rather than claimed as executed.
